### DISCUSSION TITLE
Can a node specify multiple metadata devices?

### DISCUSSION THREAD

**Discussion by microyahoo** [rook/rook#10481] [Q&A] [answered]
Can a node specify multiple metadata devices? For example, there are 20 osd disks and 2 nvme disks, and then the 2 nvme disks are divided into 20 partitions, each partition corresponds to 1 osd metadata device.

**Comment by travisn** [comment, marked as answer]
If there is a metadata device for each data device, the only way currently to configure them is with the [OSD-on-PVC approach](https://rook.io/docs/rook/latest/CRDs/ceph-cluster-crd/#dedicated-metadata-and-wal-device-for-osd-on-pvc). If using the cluster.yaml example, only a single metadata device per node can be specified.

**Reply by travisn** [reply]
See also the [example for local PVCs](https://github.com/rook/rook/blob/master/deploy/examples/cluster-on-local-pvc.yaml).

**Reply by microyahoo** [reply]
Thanks. Are there plans to support multiple metadata devices per one node?

**Reply by parth-gr** [reply]
If you think this can be great feature support to be added for you and others, 
You can create a feature request issue ticket,

Or maybe @travisn would have it's comments on it

**Reply by travisn** [reply]
@microyahoo Could you consider the approach with local PVs? This is fully flexible, so we aren't planning on adding support for it otherwise. PRs are always welcome though.

**Reply by microyahoo** [reply]
The same result can be achieved by setting `device config` to specify metadata device for every osd device, so that a node can use multiple metadata devices. https://github.com/rook/rook/blob/c8658331df09cbef3b07b1ac3f60b28267ce1635/pkg/apis/ceph.rook.io/v1/types.go#L2283
```
[root@test1 /]# lsblk | grep ceph -B 1
sdag                                                                                                   66:0    0  14.6T  0 disk 
`-ceph--ee1ad3c2--1f2b--4068--b93d--b4e8e9389e08-osd--block--e74c6106--502f--4646--b501--16c17fe167c0 253:5    0  14.6T  0 lvm  
--
sdbc                                                                                                   67:96   0  14.6T  0 disk 
`-ceph--f32f25e2--aed0--4d90--aac8--f2c70a2b2a79-osd--block--8e21e083--e021--4599--9e91--01ef1c087c3d 253:7    0  14.6T  0 lvm  
--
sdbz                                                                                                   68:208  0  14.6T  0 disk 
`-ceph--c79b8851--8590--4fea--aa70--3fe24e61e821-osd--block--aa95884b--f174--4262--bc1d--b0d5e4718989 253:1    0  14.6T  0 lvm  
--
sdcp                                                                                                   69:208  0  14.6T  0 disk 
`-ceph--f554fa30--f5a3--4bc9--8d64--39ed18dfed21-osd--block--4c10a92e--a1cb--45b9--b7f0--29bdea4cd3cd 253:3    0  14.6T  0 lvm  
--
nvme0n1                                                                                               259:0    0   1.5T  0 disk 
|-ceph--6ead527c--c299--4f39--89ac--89c50e0c810c-osd--db--29117a73--ad5b--4e5a--8cff--e6eca71cc12f    253:2    0 745.2G  0 lvm  
`-ceph--6ead527c--c299--4f39--89ac--89c50e0c810c-osd--db--dd178f46--68d1--4539--9816--6f7da6eecb1b    253:4    0 745.2G  0 lvm  
nvme1n1                                                                                               259:1    0   1.5T  0 disk 
|-ceph--e9aa11b0--f9dc--470c--92a9--ee536e0daddc-osd--db--32cb88fe--dd28--4b14--ac53--e1c26879cc66    253:6    0 745.2G  0 lvm  
`-ceph--e9aa11b0--f9dc--470c--92a9--ee536e0daddc-osd--db--fe49b70a--d0a2--47c0--a4fd--c9daf6a0b9e8    253:8    0 745.2G  0 lvm
```

**Reply by microyahoo** [reply]
[cephcluster.yaml](https://github.com/rook/rook/files/8975733/cephcluster.txt)
